terminal/src/com/jediterm/terminal/process/cache/ProcessCache.java

```diff
@@ -0,0 +1,72 @@
+package com.jediterm.terminal.process.cache;
+
+import com.jediterm.terminal.ui.UIUtil;
+import org.apache.log4j.Logger;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author gaudima
+ */
+public class ProcessCache extends Thread {
+    public interface TabNameChanger {
+        void changeName(String name);
+    }
+    protected static final Logger LOG = Logger.getLogger(ProcessCache.class);
+
+    protected Map<Integer, TabNameChanger> pidsToWatch = new ConcurrentHashMap<>();
+    protected Map<Integer, String> jobNames = new ConcurrentHashMap<>();
+    private static ProcessCache instance = null;
+    protected ProcessCache() {
+        setName("ProcessCache");
+        start();
+    }
+
+    protected String findJobName(int pid)  {
+        return "Local";
+    }
+
+    @Override
+    public void run() {
+        while (true) {
```

Comment:
If there is no tabs, here it will still spin in the cycle wasting the resources. It's better to avoid any performance overhead if terminal is closed (no tabs).